src/find/matchers/lname.rs

```diff
@@ -37,17 +37,23 @@ fn read_link_target(file_info: &DirEntry) -> Option<PathBuf> {
 /// pattern. See `glob::Pattern` for details on the exact syntax.
 pub struct LinkNameMatcher {
     pattern: Pattern,
+    follow: bool,
 }
 
 impl LinkNameMatcher {
-    pub fn new(pattern_string: &str, caseless: bool) -> LinkNameMatcher {
+    pub fn new(pattern_string: &str, caseless: bool, follow: bool) -> LinkNameMatcher {
         let pattern = Pattern::new(pattern_string, caseless);
-        Self { pattern }
+        Self { pattern, follow }
     }
 }
 
 impl Matcher for LinkNameMatcher {
     fn matches(&self, file_info: &DirEntry, _: &mut MatcherIO) -> bool {
+        // -follow: causes the -lname and -ilname predicates always to return false.
+        if self.follow {
+            return false;
```

Comment:
Not really:

```console
tavianator@graphene $ ln -s nowhere broken
tavianator@graphene $ find -lname nowhere
./broken
```

Reply:
Sorry, could you please give me some more detailed steps?
I noticed that the `-follow` parameter is missing, is that `find -follow -lname nowhere`?

Reply:
Whoops I copy pasted the wrong part of my session, but yeah:

```console
$ ln -s nowhere broken
$ find -follow -lname nowhere
./broken
```

Reply:
Thanks. I have fixed it to the actual behavior of the GNU version in https://github.com/uutils/findutils/pull/420/commits/a0d4e468676299fde7d8ae63108864a71f9770b4